storage/innobase/handler/handler0alter.cc

```diff
@@ -7695,12 +7695,15 @@ bool check_col_is_in_fk_indexes(
 
   for (const auto &a : add_fk)
   {
-    for (ulint i= 0; i < a->n_fields; i++)
+    if (dict_index_t *fk_index= a->foreign_index)
```

Comment:
This could be `const dict_index_t *`, and a similar pattern could be used elsewhere in the function, to assign `f->foreign_index` or `f->referenced_index` to a local variable.